fdbclient/MultiVersionTransaction.actor.cpp

```diff
@@ -989,64 +1013,119 @@ void MultiVersionApi::setCallbacksOnExternalThreads() {
 
 	callbackOnMainThread = false;
 }
-
 void MultiVersionApi::addExternalLibrary(std::string path) {
 	std::string filename = basename(path);
+	// we need at least one external library thread to run this library.
+	threadCount = std::max(threadCount, 1);
 
-	if(filename.empty() || !fileExists(path)) {
+	if (filename.empty() || !fileExists(path)) {
 		TraceEvent("ExternalClientNotFound").detail("LibraryPath", filename);
 		throw file_not_found();
 	}
 
-	MutexHolder holder(lock);
-	if(networkStartSetup) {
-		throw invalid_option(); // SOMEDAY: it might be good to allow clients to be added after the network is setup
-	}
-
-	if(externalClients.count(filename) == 0) {
+	if (externalClientDescriptions.count(filename) == 0) {
```

Comment:
The check whether the network has been setup has been removed, but it seems like it still doesn't work to add libraries after calling setup network. Was that intentional?

Reply:
Ok; I see.  Removing that was unintentional.  Bad merge on an intermediate commit, I think.